pkg/metrics/server.go

```diff
@@ -0,0 +1,75 @@
+package metrics
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	tlsCRT = "/etc/secrets/tls.crt"
+	tlsKey = "/etc/secrets/tls.key"
+)
+
+// BuildServer creates the http.Server struct
+func BuildServer(port int) *http.Server {
+	if port <= 0 {
+		logrus.Error("invalid port for metric server")
+		return nil
+	}
+
+	bindAddr := fmt.Sprintf(":%d", port)
+	router := http.NewServeMux()
+	router.Handle("/metrics", promhttp.Handler())
+	srv := &http.Server{
+		Addr:    bindAddr,
+		Handler: router,
+	}
+
+	return srv
+}
+
+// StopServer stops the server; for tls secret rotation
```

Comment:
the pattern we've used elsewhere is just to kill the entire operator process/pod and let it restart, when it needs to pick up a new cert.  

this isn't wrong, per se, but it might be more complicated/work than the alternative.  @ricardomaraschini recently implemented the restart on filechange logic for OCM i believe.

Reply:
yeah I was concerned about level of disruption, but not enough that I want to blaze new trails here

I'll look for @ricardomaraschini 's recent change (unfortunately I have no recollection of that PR) and go from there.

Reply:
yeah in theory rotation should not be that frequent (daily at most, more likely monthly or something) and the operator needs to be able to handle getting killed/restart anyway since that could happen for other reasons.

Reply:
fyi the change was https://github.com/openshift/cluster-image-registry-operator/pull/342/commits/ffbb403ef87ede359e941891a28378f80dfe105e

Reply:
part of next push